src/providers/cambridge/utils.py

```diff
@@ -0,0 +1,36 @@
+import requests
+import structlog
+logger = structlog.get_logger()
+
+def get_page_content(url):
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an HTTPError if the response status code indicates an error
+        return response
+    except requests.exceptions.ConnectionError as err:
+        logger.error(err)
+        return response
+    except requests.exceptions.HTTPError as err:
+        logger.error(err)
+        return response
+
+def find_links_and_tags(soup, subjects, prefix):
+    found_links = []
+    for subject in subjects:
+        target_word = prefix + subject
+        for tag in soup.find_all(string=lambda text: text and target_word in text.lower()):
+            parent_tag = tag.parent
```

Comment:
what happens if the tag doesn't have a parent? Does it return None or crash? let's say the parent tag is root?

Reply:
So should we have something like
```
            else:
                return None
```
or
```
            else:
                continue
```

Reply:
To check, please write a simple function to see what happens when:
1. Return None
2. Using else: contunue
What is the difference?

For my question, there are few ways to check it: reading docs and trying it manually:
Please try to run a small function that reads super small HTML, and try to find the parent that doesn’t  exist:
```
soup = BeautifulSoup("<p>Some</p>")
for tag in soup.find_all('p'):
    tag.parent
 ```

Reply:
@jdm010 ⬆️